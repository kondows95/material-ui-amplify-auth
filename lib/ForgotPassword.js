"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var core_1 = require("@material-ui/core");
var react_intl_1 = require("react-intl");
var FormTitle_1 = __importDefault(require("./FormTitle"));
var MyLink_1 = __importDefault(require("./MyLink"));
var material_ui_basic_parts_1 = require("material-ui-basic-parts");
var ForgotPassword = function (props) {
    var formatMessage = react_intl_1.useIntl().formatMessage;
    var _a = react_1.default.useState({ email: '', password: '' }), form = _a[0], setForm = _a[1];
    var handleChangeValue = function (fieldName) { return function (event) {
        var newForm = __assign({}, form);
        newForm[fieldName] = event.currentTarget.value;
        setForm(newForm);
    }; };
    var handleSubmit = function (event) {
        event.preventDefault();
        props.forgotPassword(form['email']);
    };
    var handleSignIn = function (event) {
        event.preventDefault();
        props.changeAuthState('signIn');
    };
    var msgId = 'MUAA.ForgotPassword';
    var msgIdTitle = msgId + '.Title';
    var msgIdSignIn = msgId + '.SignIn';
    var msgIdEmail = msgId + '.Email';
    var msgIdSubmit = msgId + '.Submit';
    var content = (react_1.default.createElement(core_1.Container, { "data-testid": msgId, component: "main", maxWidth: "xs" },
        react_1.default.createElement("form", { onSubmit: handleSubmit },
            react_1.default.createElement(core_1.Box, { display: "flex", flexDirection: "column", mt: 8 },
                react_1.default.createElement(FormTitle_1.default, { "data-testid": msgIdTitle, label: formatMessage({ id: msgIdTitle, defaultMessage: 'Reset Your Password' }) }),
                react_1.default.createElement(core_1.Box, { display: "flex", justifyContent: "center", fontWeight: 600, color: "error.main" }, props.error),
                react_1.default.createElement(core_1.Box, { width: "100%", my: 2 },
                    react_1.default.createElement(core_1.TextField, { autoComplete: "email", type: "email", onChange: handleChangeValue('email'), value: form.email, label: formatMessage({ id: msgIdEmail, defaultMessage: 'Email Address' }), variant: "outlined", required: true, fullWidth: true, inputProps: { 'data-testid': msgIdEmail } })),
                react_1.default.createElement(core_1.Box, { width: "100%", mt: 4, mb: 2 },
                    react_1.default.createElement(material_ui_basic_parts_1.MyButton, { type: "submit", label: formatMessage({ id: msgIdSubmit, defaultMessage: 'Submit' }), color: "primary", loading: props.loading, "data-testid": msgIdSubmit })),
                react_1.default.createElement(core_1.Box, { width: "100%", my: 2 },
                    react_1.default.createElement(MyLink_1.default, { label: formatMessage({ id: msgIdSignIn, defaultMessage: 'Back to Sign in' }), "data-testid": msgIdSignIn, onClick: handleSignIn }))))));
    return props.authState === 'forgotPassword' ? content : null;
};
exports.default = ForgotPassword;
