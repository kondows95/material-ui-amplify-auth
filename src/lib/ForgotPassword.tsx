import React, { MouseEvent, FormEvent, ChangeEvent } from 'react';
import { Box, Container, Link, TextField } from '@material-ui/core';
import { useIntl } from 'react-intl';
import FormTitle from './FormTitle';
import { MyButton } from 'material-ui-basic-parts';

type Props = {
    authState: string;
    changeAuthState: (state: string) => void;
    forgotPassword: (email: string) => void;
    loading: boolean;
    error: string;
};

const ForgotPassword: React.FC<Props> = (props) => {
    const { formatMessage } = useIntl();

    type Form = { [key: string]: string };
    const [form, setForm] = React.useState<Form>({ email: '', password: '' });

    type FormKey = 'email' | 'password';
    const handleChangeValue = (fieldName: FormKey) => (event: ChangeEvent<HTMLInputElement>): void => {
        const newForm = { ...form };
        newForm[fieldName] = event.currentTarget.value;
        setForm(newForm);
    };

    const handleSubmit = (event: FormEvent): void => {
        event.preventDefault();
        props.forgotPassword(form['email']);
    };

    const handleSignIn = (event: MouseEvent): void => {
        event.preventDefault();
        props.changeAuthState('signIn');
    };

    const msgId = 'MUAA.ForgotPassword';
    const msgIdTitle = msgId + '.Title';
    const msgIdSignIn = msgId + '.SignIn';
    const msgIdEmail = msgId + '.Email';
    const msgIdSubmit = msgId + '.Submit';
    const content = (
        <Container data-testid={msgId} component="main" maxWidth="xs">
            <form onSubmit={handleSubmit}>
                <Box display="flex" flexDirection="column" mt={8}>
                    <FormTitle
                        data-testid={msgIdTitle}
                        label={formatMessage({ id: msgIdTitle, defaultMessage: 'Reset Your Password' })}
                    />
                    <Box display="flex" justifyContent="center" fontWeight={600} color="error.main">
                        {props.error}
                    </Box>
                    <Box width="100%" my={2}>
                        <TextField
                            autoComplete="email"
                            type="email"
                            onChange={handleChangeValue('email')}
                            value={form.email}
                            label={formatMessage({ id: msgIdEmail, defaultMessage: 'Email Address' })}
                            variant="outlined"
                            required
                            fullWidth
                            inputProps={{ 'data-testid': msgIdEmail }}
                        />
                    </Box>
                    <Box width="100%" mt={4} mb={2}>
                        <MyButton
                            type="submit"
                            label={formatMessage({ id: msgIdSubmit, defaultMessage: 'Submit' })}
                            color="primary"
                            loading={props.loading}
                            data-testid={msgIdSubmit}
                        />
                    </Box>
                    <Box width="100%" my={2}>
                        <Link data-testid={msgIdSignIn} href="#" variant="body2" onClick={handleSignIn}>
                            {formatMessage({ id: msgIdSignIn, defaultMessage: 'Back to Sign in' })}
                        </Link>
                    </Box>
                </Box>
            </form>
        </Container>
    );

    return props.authState === 'forgotPassword' ? content : null;
};

export default ForgotPassword;
