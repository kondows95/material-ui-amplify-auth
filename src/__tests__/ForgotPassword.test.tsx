import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import TestLib from 'react-testing-library-wrapper';
import { IntlProvider } from 'react-intl';
import ForgotPassword from '../lib/ForgotPassword';

let tLib: TestLib;
let changeAuthState: jest.Mock;
let forgotPassword: jest.Mock;

beforeEach(() => {
    changeAuthState = jest.fn();
    forgotPassword = jest.fn();
});

describe('Basic Usage', () => {
    const msgs = {
        'MUAA.ForgotPassword.Title': '#MUAA.ForgotPassword.Title',
        'MUAA.ForgotPassword.Email': '#MUAA.ForgotPassword.Email',
        'MUAA.ForgotPassword.SignIn': '#MUAA.ForgotPassword.SignIn',
        'MUAA.ForgotPassword.Submit': '#MUAA.ForgotPassword.Submit',
    };
    beforeEach(() => {
        tLib = new TestLib(
            (
                <IntlProvider locale="en" messages={msgs}>
                    <ForgotPassword
                        authState={'forgotPassword'}
                        changeAuthState={changeAuthState}
                        forgotPassword={forgotPassword}
                        loading={false}
                        error={'MyError'}
                    />
                </IntlProvider>
            )
        );
    });

    it('If SignIn clicked, authState will be change.', () => {
        tLib.click('MUAA.ForgotPassword.SignIn');
        expect(changeAuthState).toHaveBeenCalledWith('signIn');
    });

    it('Click submit button', () => {
        tLib.changeValue('MUAA.ForgotPassword.Email', 'test@example.com');
        tLib.click('MUAA.ForgotPassword.Submit');
        expect(forgotPassword).toHaveBeenCalledWith('test@example.com');
    });

    it('You can set label by Intl', () => {
        const html = tLib.render.container.innerHTML;
        expect(html).toContain('#MUAA.ForgotPassword.Title');
        expect(html).toContain('#MUAA.ForgotPassword.Email');
        expect(html).toContain('#MUAA.ForgotPassword.SignIn');
        expect(html).toContain('#MUAA.ForgotPassword.Submit');
    });

    it('You can set error message', () => {
        expect(tLib.render.container.innerHTML).toContain('MyError');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});

it('Default Intl labels', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <ForgotPassword
                    authState={'forgotPassword'}
                    changeAuthState={changeAuthState}
                    forgotPassword={forgotPassword}
                    loading={true}
                    error={''}
                />
            </IntlProvider>
        )
    );
    expect(tLib.get('MUAA.ForgotPassword.Title').textContent).toBe('Reset Your Password');
    expect(tLib.get('MUAA.ForgotPassword').innerHTML).toContain('Email Address');
    expect(tLib.get('MUAA.ForgotPassword.Submit').textContent).toBe('Submit');
    expect(tLib.get('MUAA.ForgotPassword.SignIn').textContent).toBe('Back to Sign in');
});

it('If authState is not forgotPassword, return empty.', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <ForgotPassword
                    authState={'signIn'}
                    changeAuthState={changeAuthState}
                    forgotPassword={forgotPassword}
                    loading={true}
                    error={''}
                />
            </IntlProvider>
        )
    );
    expect(tLib.render.container.textContent).toBe('');
});
