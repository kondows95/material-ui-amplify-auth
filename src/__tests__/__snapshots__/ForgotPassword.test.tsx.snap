// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Usage Snapshot 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
    data-testid="MUAA.ForgotPassword"
  >
    <form>
      <div
        class="MuiBox-root MuiBox-root-682"
      >
        <div
          class="MuiBox-root MuiBox-root-683"
          data-testid="MUAA.ForgotPassword.Title"
        >
          <div
            class="MuiBox-root MuiBox-root-684"
          >
            <div
              class="MuiBox-root MuiBox-root-685"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-695"
          >
            #MUAA.ForgotPassword.Title
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-696"
        >
          MyError
        </div>
        <div
          class="MuiBox-root MuiBox-root-697"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
            >
              #MUAA.ForgotPassword.Email
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="MUAA.ForgotPassword.Email"
                required=""
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-756 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-758"
                >
                  <span>
                    #MUAA.ForgotPassword.Email *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-760"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="MUAA.ForgotPassword.Submit"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              #MUAA.ForgotPassword.Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root MuiBox-root-794"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            data-testid="MUAA.ForgotPassword.SignIn"
            href="#"
          >
            #MUAA.ForgotPassword.SignIn
          </a>
        </div>
      </div>
    </form>
  </main>
</DocumentFragment>
`;
